hibernate-reactive-core/src/main/java/org/hibernate/reactive/provider/service/ReactiveImprovedExtractionContextImpl.java

```diff
@@ -404,4 +395,1010 @@ public boolean isWrapperFor(Class<?> iface) throws SQLException {
+		}
+
+		@Override
+		public int getInt(int columnIndex) throws SQLException {
+			return delegate.getInt(columnIndex);
+		}
+
+		@Override
+		public long getLong(int columnIndex) throws SQLException {
+			return delegate.getLong(columnIndex);
+		}
+
+		@Override
+		public float getFloat(int columnIndex) throws SQLException {
+			return delegate.getFloat(columnIndex);
+		}
+
+		@Override
+		public double getDouble(int columnIndex) throws SQLException {
+			return delegate.getDouble(columnIndex);
+		}
+
+		@Override
+		public BigDecimal getBigDecimal(int columnIndex, int scale) throws SQLException {
+			return delegate.getBigDecimal(columnIndex, scale);
+		}
+
+		@Override
+		public byte[] getBytes(int columnIndex) throws SQLException {
+			return delegate.getBytes(columnIndex);
+		}
+
+		@Override
+		public Date getDate(int columnIndex) throws SQLException {
+			return delegate.getDate(columnIndex);
+		}
+
+		@Override
+		public Time getTime(int columnIndex) throws SQLException {
+			return delegate.getTime(columnIndex);
+		}
+
+		@Override
+		public Timestamp getTimestamp(int columnIndex) throws SQLException {
+			return delegate.getTimestamp(columnIndex);
+		}
+
+		@Override
+		public InputStream getAsciiStream(int columnIndex) throws SQLException {
+			return delegate.getAsciiStream(columnIndex);
+		}
+
+		@Override
+		public InputStream getUnicodeStream(int columnIndex) throws SQLException {
+			return delegate.getUnicodeStream(columnIndex);
+		}
+
+		@Override
+		public InputStream getBinaryStream(int columnIndex) throws SQLException {
+			return delegate.getBinaryStream(columnIndex);
+		}
+
+		@Override
+		public String getString(String columnLabel) throws SQLException {
+			return delegate.getString(columnLabel);
+		}
+
+		@Override
+		public boolean getBoolean(String columnLabel) throws SQLException {
+			return delegate.getBoolean(columnLabel);
+		}
+
+		@Override
+		public byte getByte(String columnLabel) throws SQLException {
+			return delegate.getByte(columnLabel);
+		}
+
+		@Override
+		public short getShort(String columnLabel) throws SQLException {
+			return delegate.getShort(columnLabel);
+		}
+
+		@Override
+		public int getInt(String columnLabel) throws SQLException {
+			return delegate.getInt(columnLabel);
+		}
+
+		@Override
+		public long getLong(String columnLabel) throws SQLException {
+			// PostgreSQL stores sequence metadata in information_schema as Strings.
+			// First try to get the value as a Long; if that fails, try to get
+			// as a String and try to parse it as a Long.
+			try {
+				return delegate.getLong(columnLabel);
+			}
+			catch (ClassCastException ex) {
+				// Check if the value is a String that can be converted to a Long
+				final String aString = delegate.getString(columnLabel);
+				// aString won't be null; check just because...
+				if (aString == null) {
+					throw ex;
+				}
+				try {
+					return Long.parseLong(aString);
+				}
+				catch (ClassCastException exNotAString) {
```

Comment:
Is this ever going to throw a `ClassCastException`?. Shouldn't it be `NumberFormatException`?

Reply:
I guess not. I mean, I just copy/pasted Gail's code here, without thinking too hard.

Reply:
OK, I fixed this. (It was a preexisting bug.)